lib/matplotlib/axes/_base.py

```diff
@@ -786,6 +786,8 @@ def _set_artist_props(self, a):
             a.set_transform(self.transData)
 
         a.axes = self
+        if a.mouseover:
+            self.mouseover_set.add(a)
```

Comment:
Alarm bells... :bell:
What if the artist is subsequently removed from the axes? This redundancy is problematic.